src/types.ts

```diff
@@ -21,3 +22,21 @@ export type User = {
   email: string;
   groupIds: string[],
 }
+
+export type Message = { 
+  createdAt: any
+  photoURL: string | null
+  text: string
+  id: string
+  name: string
+  email: string
+}
+
+export interface MessageParams extends Omit<Message, 'createdAt'> {
+  groupId: string
+}
+
+export type Chat = { 
+  groupId: string
```

Comment:
I would keep groupId out of `Chat`.

That might be part of our DB implementation, but it's not necessary for the Chat within out app  to know about our Group

Reply:
Made chats as a top-level key